### PR TITLE
3597 - update to use documents list from legal api

### PR DESCRIPTION
**Pull request by ochiu**
*Issue #:* /bcgov/entity#3597

*Description of changes:*
- update report functions to use documents list from legal-api

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/Dashboard/FilingHistoryList.vue [hunk 1]
```diff
@@ -376,6 +376,7 @@ export default {
     loadAnnualReport (filing: any) {
       const header = filing?.header
       const annualReport = filing?.annualReport
+      const documents = filing?.documents
 
       if (header && annualReport) {
```
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList.vue:381]: Should you ensure 'documents' is truthy here?

Also in other methods below.

Comment (ochiu) [reply src/components/Dashboard/FilingHistoryList.vue:381]: I do ```documents: documents || [] as Array<any>``` when assigning it to the item object, is that okay?

**Comment by ochiu** [comment]
> Well that was easy 😁

Thanks to all your hard work! It made this bit easy :)

File: src/components/Dashboard/FilingHistoryList.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/components/Dashboard/FilingHistoryList.vue:381]: You do that so the template for loop doesn't fail if there are no documents.

I'm actually wondering if 'documents' are required in all filings in the history list? I suspect they are (except for paper only).

Comment (ochiu) [reply src/components/Dashboard/FilingHistoryList.vue:381]: Currently the api will return an empty array if the logic determines there are no relevant documents to return.

Comment (severinbeauvais) [reply src/components/Dashboard/FilingHistoryList.vue:381]: Cool. So I suggest doing the same thing as with 'comments':
- don't bother ensuring it's truthy on line 380
- move the assignment on line 379 to line 398 (the filings obj) and keep your fallback safety check ([])
- do the same for the other filing types

Comment (ochiu) [reply src/components/Dashboard/FilingHistoryList.vue]: Updated. Thanks!

**Pull request merged**